src/altinn-app-frontend/src/features/form/containers/RepeatingGroupsTable.test.tsx

```diff
@@ -127,28 +145,99 @@ describe('RepeatingGroupTable', () => {
     );
   });
 
-  it('should match snapshot', () => {
-    const { asFragment } = renderWithProviders(
-      <RepeatingGroupTable
-        container={mockContainer}
-        attachments={mockAttachments}
-        language={mockLanguage}
-        textResources={mockTextResources}
-        components={mockComponents}
-        currentView={mockCurrentView}
-        editIndex={-1}
-        formData={mockData}
-        hiddenFields={[]}
-        id={mockContainer.id}
-        layout={mockLayout.layouts[mockCurrentView]}
-        options={{}}
-        repeatingGroupDeepCopyComponents={mockRepeatingGroupDeepCopyComponents}
-        repeatingGroupIndex={repeatingGroupIndex}
-        repeatingGroups={mockLayout.uiConfig.repeatingGroups}
-        setEditIndex={jest.fn()}
-        validations={{}}
-      />,
-    );
-    expect(asFragment()).toMatchSnapshot();
+  describe('desktop view', () => {
+    const { setScreenWidth } = mockMediaQuery(992);
+    beforeEach(() => {
+      setScreenWidth(1337);
+    });
+
+    it('should match snapshot', () => {
+      const { asFragment } = render();
+      expect(asFragment()).toMatchSnapshot();
+    });
+
+    it('should trigger onClickRemove on delete-button click', async () => {
+      const onClickRemove = jest.fn();
+      render({ onClickRemove: onClickRemove });
+
+      await user.click(screen.getAllByRole('button', { name: /delete/i })[0]);
+
+      expect(onClickRemove).toBeCalledTimes(1);
+    });
+
+    it('should trigger setEditIndex on edit-button click', async () => {
+      const setEditIndex = jest.fn();
+      render({ setEditIndex: setEditIndex });
+
+      await user.click(screen.getAllByRole('button', { name: /edit/i })[0]);
+
+      expect(setEditIndex).toBeCalledTimes(1);
+    });
+  });
+
+  describe('tablet view', () => {
+    const { setScreenWidth } = mockMediaQuery(992);
+    beforeEach(() => {
+      setScreenWidth(992);
+    });
+
+    it('should render as mobile-version for small screens', () => {
+      render();
+
+      const altinnMobileTable = screen.queryByTestId(/altinn-mobile-table/i);
+
+      expect(altinnMobileTable).toBeInTheDocument();
+    });
+  });
+
+  describe('mobile view', () => {
+    const { setScreenWidth } = mockMediaQuery(768);
+    beforeEach(() => {
+      setScreenWidth(768);
+    });
+
+    it('should render edit and delete buttons as icons for screens smaller thnn 786px', () => {
+      const onClickRemove = jest.fn();
+      render({ onClickRemove: onClickRemove });
+
+      const iconButtonsDelete = screen.getAllByTestId(/delete-button/i);
+      const iconButtonsEdit = screen.getAllByTestId(/edit-button/i);
+
+      expect(iconButtonsDelete).toHaveLength(4);
+      expect(iconButtonsDelete[0]).toContainHTML(
```

Comment:
What are we really interested in testing with these `toContainHTML` tests? Seems like they are almost like snapshot tests to me. Prone to breaking if we change some css classes for instance.

If you don't have a good reason for testing this I suggest just removing them.

Reply:
I wanted to test that the buttons were rendered with only the icon and no text for the mobile view. I guess a other way of doing it could be to query for elements with the button texts and assert the result to null? The main reason i added this test was for code coverage for the SonarQube gods.

Reply:
Yeah, a negative query with the button text sounds better 👍